src/base/bittorrent/peerinfo.cpp

```diff
@@ -198,7 +198,24 @@ QString PeerInfo::I2PAddress() const
 
 QString PeerInfo::client() const
 {
-    return QString::fromStdString(m_nativeInfo.client);
+    const auto source = QString::fromStdString(m_nativeInfo.client).simplified();
+    QString client;
+
+    for (const QChar &c : source)
+    {
+        // replace space-like characters with regular whitespaces
+        if (c.isSpace())
+        {
+            client.append(u' ');
+            continue;
```

Comment:
I would not abuse such things, since we have branches of equivalent "semantic weight".
```c++
if (c.isSpace())
{
    //...
}
else if (c.isPrint())
{
    //...
}
```

Reply:
I think guard clauses are better than if-else chains, but ok, it doesn't really matter.